cfme/configure/configuration/__init__.py

```diff
@@ -294,230 +301,136 @@ def step(self):
+        if self.depot_type != 'Red Hat Dropbox':
+            view.fill({'depot_name': self.depot_name,
+                       'uri': self.uri})
+        if self.depot_type in ['FTP', 'Samba']:
+            view.fill({'username': self.username,
+                       'password': self.password,
+                       'confirm_password': self.password})
+            view.validate.click()
+            view.flash.assert_success_message("Log Depot Settings were validated")
+        if cancel:
+            view.cancel.click()
+            view.flash.assert_success_message("Edit Log Depot settings was cancelled by the user")
+        else:
+            view.save.click()
+            view.flash.assert_success_message("Log Depot Settings were saved")
 
-            Args:
-                validate: Whether validate the credentials (not for NFS)
-                cancel: If set to True, the Cancel button is clicked instead of saving.
-            """
-            navigate_to(current_appliance.server, 'DiagnosticsCollectLogsEdit')
-            details = {
-                "type": sel.ByValue(self.p_types[self.p_type]),
-                "name": self.name,
-                "uri": self.uri,
-            }
-            if self.p_type not in {"nfs", "anon_ftp"}:
-                change_stored_password()
-                details["user"] = self.username
-                details["password"] = self.password
-
-            fill(
-                self.server_collect_logs,
-                details
-            )
-            if validate and self.p_type not in {"nfs", "anon_ftp", "dropbox"}:
-                sel.click(self.validate)
-                flash.assert_no_errors()
-
-            if cancel:
-                sel.click(form_buttons.cancel)
-                flash.assert_message_match("Edit Log Depot settings was cancelled by the user")
-                flash.assert_no_errors()
-            else:
-                sel.click(self.save_button)
-                flash.assert_message_match("Log Depot Settings were saved")
-                flash.assert_no_errors()
-
-        @classmethod
-        def clear(cls, cancel=False):
-            """ Navigate to correct page and set <No Depot>.
-
-            Args:
-                cancel: If set to True, the Cancel button is clicked instead of saving.
-            """
-            navigate_to(current_appliance.server, 'DiagnosticsCollectLogsEdit')
-            sel.wait_for_element(cls.server_collect_logs.type)
-            sel.wait_for_ajax()
-            if cls.server_collect_logs.type.first_selected_option_text == "<No Depot>":
-                # Nothing to do here
-                sel.click(form_buttons.cancel)
-            else:
-                fill(
-                    cls.server_collect_logs,
-                    {"type": "<No Depot>"},
-                    action=form_buttons.cancel if cancel else cls.save_button
-                )
+    @property
+    def last_collection(self):
```

Comment:
**Optional**
Consider a docstring here to define the returns - `None` if the UI displays 'never', instead of the actual string. Tests against the last collection may expect to verify against the actual UI string.